### PR TITLE
Onboarding screen shows only on first launch

### PR DESCRIPTION
**Pull request by ericlavigne**

### CODE REVIEW

File: CoEpi/domain/model/KeyValueStore.swift [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import Foundation
+
+class KeyValueStore {
+    static let boolDefaults = ["showOnboarding": true]
```
Comment (i-schuetz) [inline CoEpi/domain/model/KeyValueStore.swift:4]: We don't need to cache the values. `UserDefaults` already does this.

File: CoEpi/domain/model/KeyValueStore.swift [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    
+    static func putBool(key: String, value: Bool) {
```
Comment (i-schuetz) [inline CoEpi/domain/model/KeyValueStore.swift:6]: I'd make an enum for the keys, to make it safer. E.g:

```swift
enum PreferenceKey {
    case seenOnboardidng
}
```
See the [Android implementation](https://github.com/Co-Epi/app-android/blob/develop/app/src/main/java/org/coepi/android/system/Preferences.kt)

(You can ignore `clear` for now)

File: CoEpi/domain/model/KeyValueStore.swift [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        let defaultResult = boolDefaults[key]
+        if defaultResult == nil {
+            print("Error: Storing bool for invalid key: " + key)
+        } else {
+            UserDefaults.standard.set(value, forKey: key)
+        }
+    }
+    
+    static func getBool(key: String) -> Bool {
```
Comment (i-schuetz) [inline CoEpi/domain/model/KeyValueStore.swift:15]: I forgot to comment, don't use `static` here. The idea is to allow this component to be mocked. We can't do this with `static`.

File: CoEpi/domain/model/KeyValueStore.swift [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        let defaultResult = boolDefaults[key]
+        if defaultResult == nil {
+            print("Error: Getting bool for invalid key: " + key)
+            return defaultResult! // Will throw an error
```
Comment (i-schuetz) [inline CoEpi/domain/model/KeyValueStore.swift:19]: Let's avoid force unwrap (`!`), unless it's really unavoidable. Here it's not (this code will disappear either way if you remove `boolDefaults`. Just sayin').

File: CoEpi/domain/model/KeyValueStore.swift (hunk 1 above)
Comment (ericlavigne) [reply CoEpi/domain/model/KeyValueStore.swift]: KeyValueStore no longer specifies initial values for each of the keys. Instead, if a key's value has not yet been set, then nil will be returned.

File: CoEpi/domain/model/KeyValueStore.swift (hunk 2 above)
Comment (ericlavigne) [reply CoEpi/domain/model/KeyValueStore.swift]: Keys are now an enum.

File: CoEpi/domain/model/KeyValueStore.swift (hunk 3 above)
Comment (ericlavigne) [reply CoEpi/domain/model/KeyValueStore.swift]: No longer using static methods.

File: CoEpi/domain/model/KeyValueStore.swift (hunk 4 above)
Comment (ericlavigne) [reply CoEpi/domain/model/KeyValueStore.swift]: Force unwrap is no longer used.

File: CoEpi/domain/model/KeyValueStore.swift [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import Foundation
+
+enum KVSBoolKey: String {
```
Comment (i-schuetz) [inline CoEpi/domain/model/KeyValueStore.swift:3]: Just a minor thing, I'd not make the keys specific to the value types.

File: CoEpi/ui/RootWireframe.swift [hunk 6]
```diff
@@ -22,11 +21,12 @@ class RootWireFrame {
         window.makeKeyAndVisible()
 
         self.homeViewController = homeViewController
-
-        if showOnboarding {
+        
+        if KeyValueStore().getBool(key: KVSBoolKey.seenOnboarding) != true {
```
Comment (i-schuetz) [inline CoEpi/ui/RootWireframe.swift:25]: `KeyValueStore` should be an injectable component (this allows to change its implementation / mock it). So wire it in `Dependencies` and retrieve here with `container.resolve()` (or maybe pass it directly to the initializer, like the rest of components).

**Pull request merged**